common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -88,6 +91,16 @@ public long getLong() {
     return VariantUtil.getLong(value, pos);
   }
 
+  // Get the start and end fields of a year-month interval from the variant.
+  public IntervalFields getYearMonthIntervalFields() {
```

Comment:
What uses these apis?

Reply:
`schemaOf`, `cast from variant` and `to_json`